File: java/day-11.java
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;

public class Solution {

  public static void main(String[] args) throws IOException {
    BufferedReader bufferedReader = new BufferedReader(
      new InputStreamReader(System.in)
    );

    List<List<Integer>> arr = new ArrayList<>();

    IntStream
      .range(0, 6)
      .forEach(i -> {
        try {
          arr.add(
            Stream
              .of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
              .map(Integer::parseInt)
              .collect(toList())
          );
        } catch (IOException ex) {
          throw new RuntimeException(ex);
        }
      });

    bufferedReader.close();

    Integer maxSum = Integer.MIN_VALUE;

    for (int x = 0; x < 4; x++) {
      for (int y = 0; y < 4; y++) {
        Integer currSum = 0;

        for (int dx = 0; dx < 3; dx++) {
          for (int dy = 0; dy < 3; dy++) {
            if (dx == 1 && (dy == 0 || dy == 2)) {
              continue;
            }
            currSum += arr.get(x + dx).get(y + dy);
          }
        }

        if (currSum > maxSum) {
          maxSum = currSum;
        }
      }
    }

    System.out.println(maxSum);
  }
}

File: python/day-11.py
#!/bin/python3

import math

if __name__ == '__main__':

    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    max_sum = -math.inf
    for x in range(4):
        for y in range(4):
            curr_sum = arr[x][y] + arr[x][y+1] + arr[x][y+2] + \
                arr[x+1][y+1] + arr[x+2][y] + arr[x+2][y+1] + arr[x+2][y+2]

            if curr_sum > max_sum:
                max_sum = curr_sum

    print(max_sum)
